webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DataStructures.scala

```diff
@@ -582,37 +582,56 @@ final case class BoundPolicyDraft(
    */
   def toPolicy(agent: AgentType): Either[String, Policy] = {
     PolicyTechnique.forAgent(technique, agent).flatMap { pt =>
-      expandedVars.values.collectFirst { case v if (!v.spec.constraint.mayBeEmpty && v.values.exists(_ == "")) => v } match {
```

Comment:
here, we should test for null values too, because we have a test for null values. 
I'm almost sure we don't authorize null values since something like Rudder 2.4, but sill, I corrected the check to make it agree with the test